CRM/Contact/BAO/ContactType.php

```diff
@@ -481,6 +418,7 @@ public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
         throw new CRM_Core_Exception(ts("You can not delete this contact type -- it is used by %1 custom field group(s). The custom fields must be deleted first.", [1 => $custom->N]));
       }
     }
+    Civi::cache('contactTypes')->clear();
```

Comment:
There's already a cache clear in the post hook, but adding this extra one to the pre hook solved some issues with the unit test & might prevent other weird bugs & is otherwise harmless.